globus_cli/commands/ls.py

```diff
@@ -101,19 +38,52 @@ def ls_command(endpoint_plus_path, recursive_depth_limit,
     client = get_client()
     autoactivate(client, endpoint_id, if_expires_in=60)
 
-    # get the `ls` result
-    # note that `path` can be None
-    res = _get_ls_res(client, path, endpoint_id, recursive,
-                      recursive_depth_limit, all)
+    # create the query paramaters to send to operation_ls
+    ls_params = {"show_hidden": int(all)}
+    if path:
+        ls_params["path"] = path
 
-    def cleaned_item_name(item):
-        return item['name'] + ('/' if item['type'] == 'dir' else '')
+    # get the `ls` result
+    if recursive:
+        res = client.recursive_operation_ls(
+            endpoint_id, depth=recursive_depth_limit, **ls_params)
+    else:
+        res = client.operation_ls(endpoint_id, **ls_params)
+
+    def formated_item_name(item):
+        # determine if the formated item name should end in / or not
+        final_slash = ("/" if item["type"] == "dir" else "")
+
+        # ignore any trailing slashes for easier logic
+        if path and path[-1] == "/":
+            formatted_path = path[:-1]
+        else:
+            formatted_path = path
+
+        # if the item has a path field, it is recursive, and we want
+        # to display the relative path from the start of the ls.
+        if item.get("path"):
```

Comment:
This is a little bit nitpicky, but while we're making changes:
It would be cleaner if you did this detection based on the value of `recursive`. There's no rule that Transfer couldn't add an attribute which happens to also be named `path` in the future, and it reads a bit better/more obviously.